subiquity/common/types.py

```diff
@@ -140,11 +140,43 @@ class Bootloader(enum.Enum):
     PREP = "PREP"  # ppc64el, which puts grub on a PReP partition
 
 
+@attr.s(auto_attribs=True)
+class Partition:
+    size: int
+    number: int
+    annotations: List[str]
+
+
+@attr.s(auto_attribs=True)
+class Disk:
+    id: str
+    label: str
+    type: str
+    size: int
+    usage_labels: List[str]
+    partitions: List[Partition]
+    ok_for_guided: bool
+
+
+@attr.s(auto_attribs=True)
+class GuidedChoice:
+    disk_id: str
+    use_lvm: bool = False
+    password: Optional[str] = attr.ib(default=None, repr=False)
```

Comment:
I assume this is for password protected storage devices. How do I tell before populating a GuidedChoice whether I the disk I've selected requires a password or not?

Reply:
It's the other way around: you pass a password if you want the device to be encrypted. We don't have any support for unlocking existing dm-crypt volumes (maybe one day...)

Reply:
Ah I see!